src/goMain.ts

```diff
@@ -118,6 +126,12 @@ export function activate(ctx: vscode.ExtensionContext): void {
 				}
 			);
 
+			c.onReady().then(() => {
+				if (c.initializeResult && c.initializeResult.capabilities && !c.initializeResult.capabilities.completionProvider) {
+					ctx.subscriptions.push(vscode.languages.registerCompletionItemProvider(GO_MODE, new GoCompletionItemProvider(), '.', '\"'));
```

Comment:
We would get inside this `if` block when the user had enabled the `languageServerExperimentalFeatures['autoComplete']` setting, but the language server is old as well. In this case, we should notify the user that their version of the language server is old and does not support the `auto-complete` feature.

Same for the format feature.